publishers/HumanStatePublisher/HumanStatePublisher.cpp

```diff
@@ -250,9 +295,14 @@ void HumanStatePublisher::run()
     // PREPARE THE BASE POSITION MESSAGE
     // =================================
 
-    // Get the data from the interface
-    pImpl->humanStateBuffers.basePosition = pImpl->humanState->getBasePosition();
-    pImpl->humanStateBuffers.baseOrientation = pImpl->humanState->getBaseOrientation();
+    // Get the data from the interface if not using fixed values
+    if (!pImpl->fixBasePosition) {
```

Comment:
Instead of a separate `if` conditional check, it is better to use `else` condition in conjunction with the previous check of `fixBasePosition` or `fixBaseOrientation`

Reply:
The two options are not exclusive, it can be that we are fixing the orientation and not the position, or the other way around (e.g. what we did for the video). So the two conditions are independent.

Reply:
I mean to use 

```

if (pImpl->fixBaseOrientation) {
        pImpl->humanStateBuffers.baseOrientation[0] = fixedBaseOrientation->get(0).asFloat64();
        pImpl->humanStateBuffers.baseOrientation[1] = fixedBaseOrientation->get(1).asFloat64();
        pImpl->humanStateBuffers.baseOrientation[2] = fixedBaseOrientation->get(2).asFloat64();
        pImpl->humanStateBuffers.baseOrientation[3] = fixedBaseOrientation->get(3).asFloat64();
    }
else {
        pImpl->humanStateBuffers.basePosition = pImpl->humanState->getBasePosition();
} 

```

and similarly for `fixBaseOrientation`

Reply:
@lrapetti Sorry I was saying something stupid. The logic is correct, the if `fixBasePosition` or `fixBaseOrientation` parameters are present, the base position if updated accordingly once in the configuration and if the parameters are absent, the base position is updated on every call to the run method.